### ISSUE TITLE
Visualization

### ISSUE THREAD

**Issue by Dean-UQ** [Silverster98/HUMANISE#16]
Hi Wang,
 Nice work! I notice that you have visualized your generated human motion in .mp4 and .gif files. Since I want to consider your method as one of our baseline methods and analyze their qualitative comparisons. Can you provide the codes that visualize all generated poses in a .png file, like the following example?
![image](https://github.com/Silverster98/HUMANISE/assets/97895595/27c2ddf4-71c3-42ff-b784-8af448cb4a26)

Many Thanks!

**Comment by Silverster98** [comment]
> Hi Wang, Nice work! I notice that you have visualized your generated human motion in .mp4 and .gif files. Since I want to consider your method as one of our baseline methods and analyze their qualitative comparisons. Can you provide the codes that visualize all generated poses in a .png file, like the following example? ![image](https://user-images.githubusercontent.com/97895595/253217026-27c2ddf4-71c3-42ff-b784-8af448cb4a26.png)
> 
> Many Thanks!

Hi, I exported the per-frame body meshes of the motion sequence and used Blender to render the image/video. I manually set the view direction and the mesh color.

**Comment by Dean-UQ** [comment]
Wang, can you provide more instruction details to help me visualize the generated sequence into an image.

Many thanks for your help.

**Comment by Silverster98** [comment]
> Wang, can you provide more instruction details to help me visualize the generated sequence into an image.
> 
> Many thanks for your help.

1. Import the scene mesh into the [Blender](https://www.blender.org/).
2. Use [Stop Motion Obj](https://github.com/neverhood311/Stop-motion-OBJ) to import the human body meshes into Blender (If you want to render an image, you just need to import some specific body meshes into Blender instead of the whole sequence).
3. Set the mesh color for human body meshes.
4. Set the camera pose manually and then execute rendering.

**Comment by myy888** [comment]
Hi, in your paper, there are many generation examples with language promts like "Sit on the sofa chair that 
is close to the tv", but have you use this language prompt to generate motion? I have not seen the prompt embedding in your code

**Comment by Silverster98** [comment]
> Hi, in your paper, there are many generation examples with language promts like "Sit on the sofa chair that is close to the tv", but have you use this language prompt to generate motion? I have not seen the prompt embedding in your code

We use these language descriptions, which come from the Sr3D dataset. We use a pre-trained bert to process the descriptions, see https://github.com/Silverster98/HUMANISE/blob/main/model/basemodel.py#L35.

**Comment by myy888** [comment]
can we use language prompt like that to instruct human to behave in scenes, instead of basing on ground truth to finetune many motions? I just see, in your code related to generate, it is like "evaluation" not generation because user cannot input personal instructions.



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zan ***@***.***&gt;; 
发送时间: 2023年9月18日(星期一) 下午5:13
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Silverster98/HUMANISE] Visualization (Issue #16)





  
Hi, in your paper, there are many generation examples with language promts like "Sit on the sofa chair that is close to the tv", but have you use this language prompt to generate motion? I have not seen the prompt embedding in your code
  
We use these language descriptions, which come from the Sr3D dataset. We use a pre-trained bert to process the descriptions, see https://github.com/Silverster98/HUMANISE/blob/main/model/basemodel.py#L35.

**Comment by Silverster98** [comment]
> can we use language prompt like that to instruct human to behave in scenes, instead of basing on ground truth to finetune many motions? I just see, in your code related to generate, it is like "evaluation" not generation because user cannot input personal instructions.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Zan ***@***.***&gt;; 发送时间: 2023年9月18日(星期一) 下午5:13 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [Silverster98/HUMANISE] Visualization (Issue #16) Hi, in your paper, there are many generation examples with language promts like "Sit on the sofa chair that is close to the tv", but have you use this language prompt to generate motion? I have not seen the prompt embedding in your code We use these language descriptions, which come from the Sr3D dataset. We use a pre-trained bert to process the descriptions, see https://github.com/Silverster98/HUMANISE/blob/main/model/basemodel.py#L35. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

Do you mean that you want to develop a demo program for supporting user-given descriptions? If so, You need to modify the code to support that. The language descriptions are limited to the template presented in the paper.

**Comment by myy888** [comment]
Thanks. Another question related to the loss function. In your paper, there are two&nbsp;auxiliary losses for object grounding and action-specific generation, but where are they in your code?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Silverster98/HUMANISE"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年9月18日(星期一) 下午5:31
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Silverster98/HUMANISE] Visualization (Issue #16)





  
can we use language prompt like that to instruct human to behave in scenes, instead of basing on ground truth to finetune many motions? I just see, in your code related to generate, it is like "evaluation" not generation because user cannot input personal instructions.
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "Zan @.&gt;; 发送时间: 2023年9月18日(星期一) 下午5:13 收件人: @.&gt;; 抄送: @.&gt;; @.&gt;; 主题: Re: [Silverster98/HUMANISE] Visualization (Issue #16) Hi, in your paper, there are many generation examples with language promts like "Sit on the sofa chair that is close to the tv", but have you use this language prompt to generate motion? I have not seen the prompt embedding in your code We use these language descriptions, which come from the Sr3D dataset. We use a pre-trained bert to process the descriptions, see https://github.com/Silverster98/HUMANISE/blob/main/model/basemodel.py#L35. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: @.***&gt;
  
Do you mean that you want to develop a demo program for supporting user-given descriptions? If so, You need to modify the code to support that. The language descriptions are limited to the template presented in the paper.
